### ISSUE TITLE
Extend #395 to accommodate extra_endpoints

### ISSUE THREAD

**Issue by haidangwa** [DataDog/chef-datadog#452]
Datadog auth secrets are also stored in extra_endpoints, `node['datadog']['extra_endpoints']`, but #395 fell short in that it didn't cover these attributes. These should also be retrieved from node run state whenever available.

**Comment by olivielpeau** [comment]
This would be a welcome feature indeed. Are you still using the `extra_endpoints` on the Agent side? This was originally implemented mostly for our internal usage, but I'd be interested in your use case.

A contribution would be welcome for this feature, I'll put this in triage to estimate demand.

**Comment by haidangwa** [comment]
Yes, we are using the extra_endpoints on the agent side. In our use case,
we are in an enterprise environment where there are two disparate teams
using Datadog to monitor their area of responsibility on that particular
server. So, we have the primary team's keys configured in the main part of
the agent config, and the secondary or other teams are configured with
their keys under extra_endpoints. For example, we have a database server
with two operational teams. The first is the OS operational team, and the
second are the DBA teams. Each team needs to have their own keys in place
so they can setup monitors in their own DataDog accounts due to the
specific alerting and monitoring workflows they've already established.

On Tue, Feb 13, 2018 at 10:02 AM, Olivier Vielpeau <notifications@github.com

**Comment by olivielpeau** [comment]
Great, I see, thanks for the explanation @haidangwa!

Just as a general FYI (if someone comes across this issue), using this option without contacting our support team first incurs host double-billing (as each host will be billed for every account to which its agent reports)

**Comment by haidangwa** [comment]
@olivielpeau bump
